### ISSUE TITLE
Exception free TryNext

### ISSUE THREAD

**Issue by AJosephsen** [disruptor-net/Disruptor-net#33]
Exceptions are expensive, and should be avoided if possible.

Would it be possible to add exception free overloads

```
        bool TryNext(int n, out long next);
        bool TryNext(out long next);
```

That follows normal "TryGet" conventions

**Comment by ocoanet** [comment]
Hi,

Thank you for reporting this issue. Your are absolutely right about this : the Java code was blindly copied here but it is not following .NET conventions and, to be honest, is almost not usable.

I will add the exception free overloads.

**Comment by ocoanet** [comment]
I added the overloads. We are going to publish the changes in a new nuget package shortly.

**Comment by ocoanet** [comment]
The overloads are available in the 3.3.8 version.
